include/SH3/common/singleton.hpp

```diff
@@ -0,0 +1,79 @@
+/** @file
+ *  Singleton template class. Any object that <i>must</i> have only one instance (such as the game
+ *  engine itself, or it's specific managers [i.e scene, texture, shader]) should inherit this class.
+ *
+ *  @copyright 2016-2019 Palm Studios
+ *
+ *  @date 8-1-2019
+ *
+ *  @author Jesse Buhagiar
+ */
+
+#ifndef SINGLETON_HPP
+#define SINGLETON_HPP
+
+namespace sh3
+{
+
+template<typename T>
+class CSingleton
+{
+public:
+    typedef T SingletonType; /**< Base type of this singleton */
+
+public:
+    /**
+     * Copy constructor
+     *
+     * @note This is deleted. We do NOT want to copy a singleton under any circumstance
+     */
+    CSingleton(const CSingleton& singleton) = delete;
+
+    /**
+     * Move constructor
+     *
+     * @note This is deleted. We do NOT want to copy a singleton under any circumstance
+     */
+    CSingleton(const CSingleton&& singleton) = delete;

```

Comment:
const rvalue-references are rarely, if ever, a thing. In any case, the move constructor takes a non-const rvalue-reference.
There exists also a move-assignment operator which you can delete. Same thing with non-constness there.